test/unit/ripple/index.test.js

```diff
@@ -85,6 +93,24 @@ test('touchStart event triggers activateRipple with no onTouchStart handler', ()
   mockRaf.restore();
 });
 
+test('touchEnd event triggers deactivateRipple', () => {
+  const touchEndHandler = td.func();
+  const wrapper = mount(<DivRipple onTouchEnd={touchEndHandler}/>);
+  const foundation = wrapper.instance().foundation_;
+  foundation.deactivate = td.func();
+  wrapper.simulate('touchEnd');
+  td.verify(foundation.deactivate(td.matchers.isA(Object)), {times: 1});
+  td.verify(touchEndHandler(td.matchers.isA(Object)), {times: 1});
+});
+
+test('touchEnd event triggers deactivateRipple with no onTouchEnd handler', () => {
```

Comment:
does this test do the same thing as line 102? I think remove line 102 & line 100

Reply:
These are two different tests, one to check that the ripple deactivates when there's no handler and one for when there is. The `DivRipple` that's mounted in each of these tests are different.